### DISCUSSION TITLE
How to get the Tile currently in the viewport?

### DISCUSSION THREAD

**Discussion by Makio64** [openlayers/openlayers#16027] [General]
How to get the list of all the Tiles currently in the layer?
How to check if a tile is currently visible inside the viewport?

Context: I'm developping a new renderer based on threejs, it works but i dont see how to get the complete list of tiles.

**Comment by sweepline** [comment]
[Framestate.wantedTiles](https://openlayers.org/en/latest/apidoc/module-ol_Map.html#~FrameState) is a map of sources to the tiles that the current frame wants to use, which are the ones needed to cover the viewport. You can get them by doing something like this.
```ts
const tileSource = tileLayer.getRenderSource();
const tileSourceKey = getUid(tileSource);
const wantedTiles = frameState.wantedTiles[tileSourceKey];
```
